app/src/main/java/ch/epfl/sdp/blindly/weather/WeatherService.kt

```diff
@@ -0,0 +1,204 @@
+ */
+class DayWeather(
+    @Json(name="temp") val temperature: DayTemperature,
+    val weather: List<Weather>
+) {
+}
+
+/**
+ * The weather for the following week (7 days)
+ *
+ * @property daily the weather for a day
+ */
+class WeekWeather(val daily: Array<DayWeather>) {
+}
+/**
+ * The weather during some time
+ *
+ * @param description the textual description of the weather
+ */
+class Weather(val description: String, private val icon: String) {
+    companion object {
+        private val ICON_MAP = mapOf(
+            "01n" to R.drawable.weather_01n,
+            "02n" to R.drawable.weather_02n,
+            "03n" to R.drawable.weather_03n,
+            "04n" to R.drawable.weather_04n,
+            "09n" to R.drawable.weather_09n,
+            "10n" to R.drawable.weather_10n,
+            "11n" to R.drawable.weather_11n,
+            "13n" to R.drawable.weather_13n,
+            "50n" to R.drawable.weather_50n,
+            )
+    }
+
+    fun getIcon() = ICON_MAP[icon]
+
+}
+
+/**
+ * Service to get the weather
+ *
+ */
+class WeatherService {
+    private companion object {
+        private const val url = BuildConfig.OPEN_WEATHER_MAP_URL
+        private const val openWeatherMapAppId = BuildConfig.OPEN_WEATHER_MAP_KEY
+        private const val DEFAULT_LANG = "en"
+        private val SUPPORTED_LANGUAGES = arrayOf(
+            "af",
+            "al",
+            "ar",
+            "az",
+            "bg",
+            "ca",
+            "cz",
+            "da",
+            "de",
+            "el",
+            "en",
+            "eu",
+            "fa",
+            "fi",
+            "fr",
+            "gl",
+            "he",
+            "hi",
+            "hr",
+            "hu",
+            "id",
+            "it",
+            "ja",
+            "kr",
+            "la",
+            "lt",
+            "mk",
+            "no",
+            "nl",
+            "pl",
+            "pt",
+            "pt_br",
+            "ro",
+            "ru",
+            "sv",
+            "se",
+            "sk",
+            "sl",
+            "sp",
+            "es",
+            "sr",
+            "th",
+            "tr",
+            "ua",
+            "uk",
+            "vi",
+            "zh_cn",
+            "zh_tw",
+            "zu"
+        )
+    }
+
+    private var client: OkHttpClient = OkHttpClient()
+
+    abstract class WeatherResultCallback {
+        abstract fun onFailure(e: Exception)
+        abstract fun onResponse(weather: WeekWeather)
+    }
+    private fun getLanguage(): String {
+        var language = Locale.getDefault().displayLanguage
+        if (SUPPORTED_LANGUAGES.contains(language))
+            language = DEFAULT_LANG
+        return language
+    }
+    fun nextWeek(
+        loc: BlindlyLatLng,
+        language: String = getLanguage(),
+        callback: WeatherResultCallback
+    ) {
+        val u: HttpUrl? =  url.toHttpUrlOrNull()
+        val url = u!!.newBuilder()
```

Comment:
On line 166,  if url is valid ```url``` is converted to http or null as ```u``` , but on line 167 a non-null assertion is put for ```u!!```. It might cause some issues if the url is not valid right?

Reply:
Yes, in fact I use the `toHttpUrlOrNull` as it is the only non-deprecated method to do so, however as the url is a compilation constant it will never be invalid if the tests passed, so as long as we run the tests before publishing anything it will always works.
Nevertheless I added a comment about this.

Reply:
Great! Thanks for taking the time to do so.